packages/modules/viewer-react/src/types.ts

```diff
@@ -113,7 +142,7 @@ export interface LoaderProps {
   loadingComponent?: React.ReactNode;
 }
 
-export type ViewerCommonProps = ViewerInitializerParams & LoaderProps & UnifiedSelectionProps;
+export type ViewerCommonProps = ViewerInitializerParams & LoaderProps & (UnifiedSelectionProps | { [Property in keyof UnifiedSelectionProps]?: never });
```

Comment:
Perhaps extract this type since it's a bit unclear at first glance. Maybe an `AllOrNone<T>` utility type?

Reply:
1. Extracted selection scope from `UnifiedSelectionProps` into a separate `SelectionScopeProps` type.
2. Replaced usages of `UnifiedSelectionProps | { [Property in keyof UnifiedSelectionProps]?: never }` with `AllOrNone<UnifiedSelectionProps>`.
3. Moved `UnifiedSelectionProps` from `ViewerCommonProps` and `ModelLoaderProps` into `LoaderProps`, which is base for both of them.

Reply:
@ben-polinsky mind resolving the comment and approving the PR if you got no other suggestions?